app/states/unified_database.py

```
@@ -0,0 +1,1208 @@
+                elif isinstance(value, datetime):
+                    data_dict[k] = value.isoformat()
+                elif isinstance(value, Decimal):
+                    data_dict[k] = float(value)
+                else:
+                    data_dict[k] = value
+            
+            version = UnifiedTransactionVersion(
+                transaction_id=tx.id,
+                version_number=version_count + 1,
+                data=json.dumps(data_dict),
+                changed_at=datetime.utcnow(),
```

Ruff (0.14.7)

`datetime.datetime.utcnow()` used

(DTZ003)